File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int i=0;
        int j=0;
        int[] freq=new int[256];
        int len=s.length();
        char[] str=new char[len];
        int ans=Integer.MIN_VALUE;
        int sum=0;
        for(int k=0;k<len;k++)
        {
            str[k]=s.charAt(k);
        }
        while(j<len)
        {
            sum+=str[j];
            freq[str[j]]++;
            while(freq[str[j]]>2)
            {
                freq[str[i]]--;
                i++;
            }
            ans=Math.max(ans,j-i+1);
            j++;
        }
        return ans;
    }
}

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        i=0
        j=0
        freq={}
        ans=0
        while(j<len(s)):
            freq[s[j]]=freq.get(s[j],0)+1
            while(freq[s[j]]>2):
                freq[s[i]]-=1
                i+=1
            ans=max(ans,j-i+1)
            j+=1
        return ans